BeeJet.Storage/Databases/SteamIdDiscordUserDb.cs

```diff
@@ -0,0 +1,50 @@
+﻿using BeeJet.Storage.Entities;
+using BeeJet.Storage.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BeeJet.Storage.Databases
+{
+    internal class SteamIdDiscordUserDb : BaseDb<ISteamIdDiscordUser, SteamIdDiscordUser>, ISteamIdDiscordUserDb
+    {
+        public SteamIdDiscordUserDb(ILiteDatabase database) : base(database)
+        {
+        }
+
+        protected override void EnsureIndexes()
+        {
+            Collection.EnsureIndex(b => b.Id);
+            Collection.EnsureIndex(b => b.SteamId);
+            Collection.EnsureIndex(b => b.DiscordId);
+        }
+
+        protected override string ProvideCollectionName() => "steamdiscordmappings";
+
+        public void AddOrUpdateMapping(string discordId, string steamId)
```

Comment:
Ook hier zou ik denk ik geen Or in de naam doen en zou de check hier buiten doen.
Omdat dit 2 dingen doet.
Je zou ook bijvoorbeeld volgende kunnen doen.
DetermineMapping
als die iets oplevert dan UpdateMapping en anders AddMapping.

Reply:
Op zich heb je technisch gezidn gelijk maar in praktische zin doet wel maar één ding: een mapping aanmaken. Hij kan er nog niet zijn en dan is het add, en anders wordt het een update. Maar het eindresultaat is altijd hetzelfde. Je zou ook kunnen stellen dat de methodenaam gewoon 'te beschrijvend' was en informatie gaf die niet noodzakelijk was voor de buitenwereld.

`SetMapping` of `AssignMapping` had in mijn ogen ook gekund. Deze methode zie ik zoals het setten van een dictionary value o.b.v. een key. Dat is in mijn ogen ook geen SRP overtreding :)

Reply:
Ik heb er inderdaad `SetDiscordSteamMapping` van gemaakt

Reply:
Ik had eerst je commit moeten bekijken haha, ik zag het nu ook 😅 maar dit lijkt me prima zo idd. Aan de binnenkant wel opgebroken maar voor de public zijde gewoon één ingang :)